utils.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Description: Common helper functions
+
+import hashlib
+import pathlib
+import shutil
+import subprocess
+
+
+def create_gitignore(folder):
+    """
+    Create a gitignore that ignores all files in a folder. Some folders are not
+    known until the script is run so they can't be added to the root .gitignore
+    :param folder: Folder to create the gitignore in
+    """
+    with folder.joinpath(".gitignore").open("w") as gitignore:
+        gitignore.write("*")
+
+
+def current_binutils():
+    """
+    Simple getter for current stable binutils release
+    :return: The current stable release of binutils
+    """
+    return "binutils-2.32"
+
+
+def download_binutils(root):
+    """
+    Downloads the latest stable version of binutils
+    :param root: Directory to download binutils to
+    """
+    binutils = current_binutils()
+    p = pathlib.Path.joinpath(root, binutils)
```

Comment:
This and the rest could be `root / binutils`, if you like to be fancy.

Reply:
I personally prefer the `joinpath` method but if others prefer this, I can certainly change it.